### PR TITLE
Updates for Polymer 3. (v4.0.0 prereleases)

### PR DESCRIPTION
**Pull request by bicknellr**
(I'll tag and release v4.0.0 after this is in.)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
+    "@polymer/iron-selector": "^3.0.0-pre.15",
+    "@polymer/paper-icon-button": "^3.0.0-pre.15",
+    "@polymer/polymer": "^3.0.0",
+    "@webcomponents/webcomponentsjs": "^2.0.0"
+  },
+  "devDependencies": {
+    "polymer-cli": "^1.7.0",
+    "wct-browser-legacy": "^1.0.0"
+  },
+  "engines": {
```
Comment (keanulee) [inline package.json]: Do we need `"engines"`?

File: src/my-icons.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import '@polymer/iron-iconset-svg/iron-iconset-svg.js';
+const $_documentContainer = document.createElement('div');
```
Comment (keanulee) [inline src/my-icons.js]: These should be `<template>`s now (and `.appendChild($_documentContainer.content);` below) due to template polyfill limitations on IE. See https://github.com/Polymer/shop/blob/3.0/src/shop-form-styles.js

File: src/my-app.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        type: String,
+        reflectToAttribute: true,
+        observer: '_pageChanged'
+      },
+      routeData: Object,
+      subroute: Object
+    };
+  }
+
+  ready() {
```
Comment (abdonrd) [inline src/my-app.js]: We can remove it.

**Comment by bicknellr** [comment]
I *think* we're going to go ahead with 4.0.0 even though the elements are still in prerelease, but I'm double checking with some other folks right now to make sure.

File: package.json (hunk 1 above)
Comment (bicknellr) [reply package.json]: I think this was added because the CLI doesn't support Node.js <8 any more.

File: src/my-icons.js (hunk 2 above)
Comment (bicknellr) [reply src/my-icons.js]: Thanks for pointing this out; fixed!

File: package.json [hunk 4]
```diff
@@ -1,15 +1,34 @@
...
+    "@polymer/iron-iconset-svg": "^3.0.0-pre.15",
+    "@polymer/iron-media-query": "^3.0.0-pre.15",
+    "@polymer/iron-pages": "^3.0.0-pre.15",
+    "@polymer/iron-selector": "^3.0.0-pre.15",
+    "@polymer/paper-icon-button": "^3.0.0-pre.15",
+    "@polymer/polymer": "^3.0.0",
+    "@webcomponents/webcomponentsjs": "^2.0.0"
+  },
+  "devDependencies": {
+    "polymer-cli": "^1.7.0",
```
Comment (keanulee) [inline package.json]: Should use 1.7.1 which significantly reduces deps required.

**Pull request merged**